js/client/modules/@arangodb/testutils/process-utils.js

```diff
@@ -322,6 +322,14 @@ function setupBinaries (builddir, buildType, configDir) {
       isEnterpriseClient = true;
       checkFiles.push(ARANGOBACKUP_BIN);
     }
+    if (version.asan &&
+        !process.env.hasOwnProperty('UBSAN_OPTIONS') &&
+        fs.exists('ubsan_arangodb_suppressions.txt')
+       ) {
+      print('preparing asan environment');
+      process.env['UBSAN_OPTIONS'] = `suppressions=${fs.join(fs.makeAbsolute(''), 'ubsan_arangodb_suppressions.txt')}`;
+      process.env['TSAN_OPTIONS'] = `suppressions=${fs.join(fs.makeAbsolute(''), 'tsan_arangodb_suppressions.txt')}`;
```

Comment:
Right now the code in this PR may work, but it is not consistent.

It checks for the non-presence of the `UBSAN_OPTIONS` env variable and for the file presence of `ubsan_arangodb_suppressions.txt`, and then adds the `UBSAN_OPTIONS` env variable pointing to the file, which looks ok.

But why does it set the `TSAN_OPTIONS` env variable pointing to another file, which may or may not exist? Why not apply the same check here for `TSAN_OPTIONS` and check if the tsan suppressions file exists? In addition, it could honor the `version.tsan` flag in this case, which already exists.

Finally,why not do the same for `ASAN_OPTIONS` and the asan suppressions file?

Apart from that, you cannot enable ASan and TSan at the same time. From man g++:

>        -fsanitize=thread
>           Enable ThreadSanitizer, a fast data race detector.  Memory access instructions are instrumented to detect data race bugs.  ...
>           The option cannot be combined with -fsanitize=address, -fsanitize=leak.

So I suggest there should be 3 separate checks for ASAN, UBSAN and TSAN here, instead of just one.